en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
+- Note there is only a minimal performance penalty for this, though there is a downside that IDE type hinting stops working. During development you may wish to use the new keyword for this reason and change it to use injector before submitting your PR.
+- For classes that use the Injectable trait use the `::create()` static method.
+- For objects that do not have the Injectable trait, use `Injector::inst()->get({classname})`
+- This also includes instantiation of third party classes `<<< @todo why? discuss`
+- The exception to this is for unit-tests where you should use the `new` because there you need to ensure you're testing a specific class
+
+## Extensions and traits
+
+- Use the DataExtension class for extending DataObjects.
+- Use the Extension class for extending non-DataObjects.
+- Typically you shouldn't use native PHP traits. A common gotcha is mistakenly using a trait on a DataObject to add a private static config property. This doesn't work because private static config properties are a Silverstripe specific thing and the Silverstripe DataExtension will correctly merge them. However traits are unable to merge private static config properites.
+
+### Extension hooks
+
+- Naming convention for extension hooks is `update{methodName}` where methodName has any leading `get` omitted has the remaining first letter uppercased.
+
+### Unit testing
+
+- The unit test method name should match the method that it's testing with the first letter of the method uppercased and the word `test` prefixed e.g. for the method `myMethodName()` the unit test method should be called `testMyMethodName()`
+- Aim to use DataProvider's to provide test case data as it makes code much cleaner and it makes it very easy to add additional test cases.
+- The DataProvider method name should be same as the test case method name it's providing for with the leading work `test` substituted for `provide` e.g. for `testSomething()` the DataProvider is `provideSomething()`
+- Typically you should use yml fixture files for creating fixtures, though this is a fairly loose recommendation. Often it will make sense to "create them manually" in PHP.
+- Using reflection to change the visibility of a private or protected method so that it can be independently tested is usually OK. At a practical level sometimes only testing public methods is overly clunky.
+
+### Admin-only controllers
+
+- Controllers that are only intended to be used from within the CMS should extend [`LeftAndMain`](api:SilverStripe\Admin\LeftAndMain). This will handle routing for you and ensures that only authenticated users are able to access their endpoints.
+- For small modules that are expected to only have a single admin-only controller, the `$url_segment` private static configuration property should match the second half of the composer name of the module e.g. for a `LinkFieldController` in `silverstripe/linkfield` the controllers `$url_segment` should be "linkfield".
```

Comment:
I don't think I agree with this. The url segment should be something meaningful for the controller in question. If the controller in question does interact with the actual link _field_ then yes that makes sense. If the linkfield controller was actually a model admin for managing links though, `linkfield` might not be the most appropriate URL segment.

I think we can leave this out as far as standards go and let the developer and peer reviewer come to a consensus about an appropriate URL segment on a per-controller basis.

Reply:
Another reason not to do this is just because we don't expect a second controller for that module _now_ doesn't mean it won't ever have one. If we're only using the "linkfield" segment because of this standard, and then later we add a second controller that would be _way_ more appropriate to use that segment, then we've painted ourselves into a corner.

Reply:
Yeah fair, this probably not the right place to come up with a definitive guide to CMS URL standards.

Reply:
Since we're literally in the process of coming up with a new way of doing this, maybe we skip this one for now?

Reply:
Have removed